assets/javascripts/discourse/components/activity-pub-tag-chooser.js

```diff
@@ -0,0 +1,54 @@
+import { computed } from "@ember/object";
+import I18n from "I18n";
+import TagDrop, {
+  ALL_TAGS_ID,
+  NO_TAG_ID,
+  NONE_TAG,
+} from "select-kit/components/tag-drop";
+import ActivityPubActor from "../models/activity-pub-actor";
+
+export default TagDrop.extend({
+  classNames: ["activity-pub-tag-chooser"],
+
+  content: computed("topTags.[]", function () {
+    return this.activityPubFilter(this.topTags);
+  }),
+
+  activityPubFilter(tags) {
+    return tags.filter((tag) => {
+      if (tag === this.tagId) {
+        return false;
+      }
+      const actor = ActivityPubActor.findByModel(tag, "tag");
+      if (this.selectKit.options.hasActor) {
+        return !!actor;
+      } else {
+        return !actor;
+      }
+    });
+  },
+
+  search(filter) {
```

Comment:
The filtering in the "Select a tag" dropdown doesn't seem to be working.

Reply:
@pmusaraj I've added some fixes and written unit tests for the component: https://github.com/angusmcleod/discourse-activity-pub/commit/bc700855abec7b30c3ae4102fa1a400922d48418